pkg/subscraping/agent_test.go

```
@@ -0,0 +1,132 @@
+				chunk = remaining
+			}
+			nw, werr := w.Write(buf[:chunk])
+			if werr != nil {
+				return
+			}
+			remaining -= nw
+		}
+	}))
+	defer server.Close()
+
+	req, err := http.NewRequest(http.MethodGet, server.URL, nil)
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)